docs/source/plugin/test.rst

```diff
@@ -0,0 +1,323 @@
+:func:`~sopel.tests.pytest_plugin.botfactory` fixture for that::
+
+    def test_my_command(tmpconfig, botfactory):
+        bot = botfactory(tmpconfig)
+
+However at this point, the bot doesn't know about your plugin, so it hasn't
+run the setup phase. You can do that with the
+:meth:`~sopel.tests.factories.BotFactory.preloaded` method::
+
+    def test_my_command(tmpconfig, botfactory):
+        bot = botfactory.preloaded(tmpconfig, ['myplugin'])
+        assert bot.has_plugin('myplugin')  # should be True
+
+.. note::
+
+    When using the :meth:`~sopel.tests.factories.BotFactory.preloaded` method,
+    you must not define your config sections manually, as this should be done
+    by your ``setup`` plugin hook.
+
+Of course, if you want to reuse the same test bot in all your tests, you can
+create a fixture for that::
+
+    @pytest.fixture
+    def bot(configfactory, botfactory):
+        settings = configfactory(TEST_NAME, TEST_CONFIG)
+        return botfactory.preloaded(settings, ['myplugin'])
+
+And then use it in your tests::
+
+    def test_my_command(bot):
+        assert bot.has_plugin('myplugin')  # should be True
+
+The ``bot`` created by the factory is a regular instance of
+:class:`sopel.bot.Sopel` with a
+:class:`test IRC backend <sopel.tests.mocks.MockIRCBackend>` instead of a
+regular one. This backend doesn't send anything over the network and instead
+it registers everything into its ``message_sent`` list::
+
+    from sopel.tests import rawlist
+
+
+    def test_my_command(bot):
+        bot.say('Hi!', '#channel')
+        assert len(bot.backend.message_sent) == 1
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG #channel :Hi!',
+        )
+
+.. seealso::
+
+    For more information about the :func:`~sopel.tests.rawlist` function,
+    see the `Checking the bot's output`_ section.
+
+Test Server and test Users
+--------------------------
+
+Alright, you have a properly setup test bot, now it's time for the last parts
+of your test setup: a test server with test users. As usual, there are fixtures
+to help you. The :func:`~sopel.tests.pytest_plugin.ircfactory` can be used to
+create a test server, and the :func:`~sopel.tests.pytest_plugin.userfactory`
+can create test users::
+
+    @pytest.fixture
+    def irc(bot, ircfactory):
+        return ircfactory(bot)
+
+    def test_my_command(bot, irc, userfactory):
+        user = userfactory('MyNick')
+        irc.pm(user, '.mycommand arg')
+
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG MyNick :Command answer to a regular user.'
+        )
+
+    def test_my_command_owner(bot, irc, userfactory):
+        owner = userfactory('OwnerNick')
+        irc.pm(owner, '.mycommand arg')
+
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG MyNick :Command answer to my owner.'
+        )
+
+As usual, you can create custom fixtures for the test server and for your test
```

Comment:
```suggestion
As usual, you can create custom fixtures for the test server (as above) and for your test
```

Because it's been shown already, no need to repeat. But refer to the above example in case the user glosses over that part.

Reply:
See, that's why everyone needs an editor!